runtime/realm/inst_layout.h

```diff
@@ -144,7 +144,7 @@ namespace Realm {
     bool serialize(S& serializer) const;
 
     ZPoint<N, size_t> strides;
-    size_t offset;
+    ptrdiff_t offset;
```

Comment:
@eddy16112 I know it looks weird, but this does need to be `size_t`.  Over/underflow is not defined for signed integers (dumb, but true).

Reply:
what if the offset is negative? I know the size_t is able to handle this kind of underflow, and somehow the entire program still works, but I am not sure if it will bring any invisible bugs.

Reply:
A negative offset becomes a very large size_t value which, when added to another size_t value, wraps around 2^64 correctly and results in a sum that is the right amount smaller.